src/test/java/com/alipay/remoting/ConnectionEventListenerTest.java

```diff
@@ -0,0 +1,46 @@
+package com.alipay.remoting;
+
+import com.alipay.remoting.rpc.RpcClient;
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+
+import static org.junit.Assert.*;
+
+public class ConnectionEventListenerTest {
+
+    @Test
+    public void addConnectionEventProcessorConcurrentTest() throws InterruptedException {
```

Comment:
This unit test can not reproduce the bug. 
1. `new Thread(thread).start() `: Thread is started in order. So, `rpcClient.addConnectionEventProcessor(ConnectionEventType.CONNECT, (remoteAddress, connection) -> {});
` may called in order. We can add a CountDownLatch before `addConnectionEventProcessor` method, so `addConnectionEventProcessor` can be concurrency.
2.  `fail()` method is called in sub thread . We can not see it in main thread.
3. ArrayList maybe thread safe for add method.

Reply:
look at issue "Steps to reproduce"

Reply:
When it change 'CopyOnWriteArrayList' to 'ArrayList', this unit test cannot  throw any Exception.So I say this unit test can not reproduce the bug. Please pay attention to it. @chuailiwu @EvenLjj

Reply:
This unit test indeed can not reproduce the bug, even I add a CyclicBarrier to control all threads run in same time.
But the fix is right, there is a concurrency issue here.
![image](https://user-images.githubusercontent.com/2288906/233595655-38d83c9f-11b0-41b5-9d43-bbf66eab25a6.png)

Reply:
Yes. I tried use arrayList to reproduce the bug, but failed when there is only  `add operation`.  When I add `remove operation` in ut, it can reproduce similar problem. 

Now the question become how user trigger the  bug. It may be user is traversing the list, at the same time , he add processor to the list.  Is it possible? or we can do something else to forbid it ? (I think it is not right to add processor when other processors are working  on connections.It may cause different behavior in different connections. )